qiskit/quantum_info/analysis/z2_symmetries.py

```diff
@@ -0,0 +1,471 @@
+        ret.append("Qubit index:")
+        ret.append(str(self._sq_list))
+        ret.append("Tapering values:")
+        if self.tapering_values is None:
+            possible_values = [
+                str(list(coeff)) for coeff in itertools.product([1, -1], repeat=len(self._sq_list))
+            ]
+            possible_values = ", ".join(x for x in possible_values)
+            ret.append("  - Possible values: " + possible_values)
+        else:
+            ret.append(str(self.tapering_values))
+
+        ret = "\n".join(ret)
+        return ret
+
+    def is_empty(self) -> bool:
+        """
+        Check the z2_symmetries is empty or not.
+
+        Returns:
+            Empty or not.
+        """
+        return len(self._symmetries) == 0 or len(self._sq_paulis) == 0 or len(self._sq_list) == 0
+
+    # pylint: disable=invalid-name
+    @classmethod
+    def find_Z2_symmetries(cls, operator: SparsePauliOp) -> Z2Symmetries:
+        """
+        Finds Z2 Pauli-type symmetries of a :class:`.SparsePauliOp`.
+
+        Returns:
+            A ``Z2Symmetries`` instance.
+        """
+        pauli_symmetries = []
+        sq_paulis = []
+        sq_list = []
+
+        stacked_paulis = []
+
+        if operator.is_zero():
+            logger.info("Operator is empty.")
+            return cls([], [], [], None)
+
+        for pauli in operator:
+            stacked_paulis.append(
+                np.concatenate((pauli.paulis.x[0], pauli.paulis.z[0]), axis=0).astype(int)
+            )
+
+        stacked_matrix = np.stack(stacked_paulis)
+        symmetries = _kernel_F2(stacked_matrix)
+
+        if not symmetries:
+            logger.info("No symmetry is found.")
+            return cls([], [], [], None)
+
+        stacked_symmetries = np.stack(symmetries)
+        symm_shape = stacked_symmetries.shape
+        half_symm_shape = symm_shape[1] // 2
+        stacked_symm_del = [
+            np.delete(stacked_symmetries, row, axis=0) for row in range(symm_shape[0])
+        ]
+
+        def _test_symmetry_row_col(row: int, col: int, idx_test: list, row_test: list) -> bool:
+            """
+            Utilitary method that determines how to build the list of single-qubit Pauli X operators and
+            the list of corresponding qubit indices from the stacked symmetries.
+            This method is successively applied to Z type, X type and Y type symmetries (in this order)
+            to build the letter at position (col) of the Pauli word corresponding to the symmetry at
+            position (row).
+
+            Args:
+                row (int): Index of the symmetry for which the single-qubit Pauli X operator is being
+                    built.
+                col (int): Index of the letter in the Pauli word corresponding to the single-qubit Pauli
+                    X operator.
+                idx_test (list): List of possibilities for the stacked symmetries at all other rows
+                    than row and.
```

Comment:
Did you mean to add something here or is the trailing `and` a leftover?

Reply:
No it was a typo.